sunpy/net/dataretriever/sources/rhessi.py

```diff
@@ -151,65 +151,23 @@ def get_observing_summary_dbase_file(time):
         url = get_base_url() + f'dbase/{_time.strftime("hsi_obssumm_filedb_%Y%m.txt")}'
         return urlretrieve(url)
 
-    def _get_url_for_timerange(self, timerange, **kwargs):
-        """
-        Returns a URL to the RHESSI data for the specified date range.
-
-        Parameters
-        ----------
-        timerange : `~sunpy.time.TimeRange`
-            Date range should be specified using a TimeRange.
-        """
-        return self.get_observing_summary_filename(timerange)
-
-    def _get_time_for_url(self, urls):
-        ts = [datetime.strptime(url.split("hsi_obssumm_")[1].split("_")[0],
-                                "%Y%m%d") for url in urls]
-        return [TimeRange(t, (1*u.day-1*u.ms)) for t in ts]
-
-    def _makeimap(self):
-        """
-        Helper Function:used to hold information about source.
-        """
-        self.map_['source'] = 'rhessi'
-        self.map_['instrument'] = 'rhessi'
-        self.map_['physobs'] = 'irradiance'
-        self.map_['provider'] = 'nasa'
-
-    @classmethod
-    def _can_handle_query(cls, *query):
-        """
-        Answers whether client can service the query.
-
-        Parameters
-        ----------
-        query : list of query objects
-
-        Returns
-        -------
-        boolean
-            answer as to whether client can service the query
-        """
-        from sunpy.net import attrs as a
-
-        required = {a.Time, a.Instrument}
-        optional = {a.Physobs}
-        if not cls.check_attr_types_in_query(query, required, optional):
-            return False
-
-        matches = True
-        for x in query:
-            if isinstance(x, a.Instrument) and x.value.lower() != 'rhessi':
-                matches = False
-            if isinstance(x, a.Physobs) and x.value.lower() != 'summary_lightcurve':
-                matches = False
-
-        return matches
+    def search(self, *args, **kwargs):
+        baseurl, pattern, matchdict = self.pre_search_hook(*args, **kwargs)
+        timerange = matchdict['Time']
+        metalist = []
+        for url in self.get_observing_summary_filename(timerange):
+            exdict = parse(pattern, url).named
+            exdict['url'] = url
+            map_ = super().post_search_hook(exdict, matchdict)
+            metalist.append(map_)
```

Comment:
Is `map_` a good name for this variable?

Reply:
Changed to `rowdict` now everywhere.